### PR TITLE
Dijkstra planner readme

### PR DESCRIPTION
**Pull request by orduno**
Adding information about the Djkstra Planner.

### CODE REVIEW

File: nav2_dijkstra_planner/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The DijkstraPlanner implements the Task Server interface for ComputePathToPose, specifically derives from `ComputePathToPoseTaskServer`. The client to DijkstraPlanner is 'NavigateToPoseTask', which periodically sends requests to the path planner.
+
+![alt text](../doc/design/NavigationSystemTasks.png "Navigation Tasks")
+
+## Testing
+Currenty, component level tests are available that exercise the Planning Module Task Server API. The intention is to develop standard tests that can be used for any type of planner.
+
+A PlannerTester node provides the world representation in the form of a costmap, sends a request to generate a path, and receives and checks the quality of the generated path.
+
```
Comment (SteveMacenski) [inline nav2_dijkstra_planner/README.md]: I'm not sure that the testing explanation should overshadow the actual package's description. To be fair, its a really neat picture you generated, but I might just follow the KISS principle here. Probably involving removing all the text (lines 25-33) below this comment

File: nav2_dijkstra_planner/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Currently, DijkstraPlanner is a direct port from ROS1 MoveBase [Navfn](http://wiki.ros.org/navfn) planner.
+
+The Navfn planner assumes a circular robot and operates on a costmap to find a minimum cost plan from a start point to an end point in a grid. The navigation function is computed with Dijkstra's algorithm. The strategy is based on [High-Speed Navigation Using
+the Global Dynamic Window Approach](https://cs.stanford.edu/group/manips/publications/pdfs/Brock_1999_ICRA.pdf) Brock, O. and Oussama K. IEEE (1999)
+
+## Task Interface
+
+The [Navigation System]((../doc/requirements/requirements.md)) is composed of three tasks: NavigateToPose, ComputePathToPose and FollowPathToPose.
+
+The DijkstraPlanner implements the Task Server interface for ComputePathToPose, specifically derives from `ComputePathToPoseTaskServer`. The client to DijkstraPlanner is 'NavigateToPoseTask', which periodically sends requests to the path planner.
```
Comment (SteveMacenski) [inline nav2_dijkstra_planner/README.md]: > `nav2_tasks::ComputePathToPoseTaskServer`

its useful to tell someone where to find this base class

File: nav2_dijkstra_planner/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Dijkstra Planner
```
Comment (SteveMacenski) [inline nav2_dijkstra_planner/README.md]: I might add a blurb explaining why / when you should / would want to use this over an A* planner.

File: nav2_dijkstra_planner/README.md (hunk 1 above)
Comment (orduno) [reply nav2_dijkstra_planner/README.md]: Agree, this explanation should be removed from this package description and moved to system tests.

Comment (SteveMacenski) [reply nav2_dijkstra_planner/README.md]: Or even the tests as well moved over there

File: nav2_dijkstra_planner/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+The DijkstraPlanner is an implementation of a [Planning module](../doc/requirements/requirements.md) (equivalent to GlobalPlanner in ROS1 [MoveBase](http://wiki.ros.org/move_base)).
```
Comment (mjeronimo) [inline nav2_dijkstra_planner/README.md]: I'd say that the DijkstraPlanner is a planning module that implements the ComputePathToPose interface.

File: nav2_dijkstra_planner/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+The Planning module is responsible for generating a feasible path given start and end robot poses.
```
Comment (mjeronimo) [inline nav2_dijkstra_planner/README.md]: "A planning module implementing the ComputePathToPose interface is responsible..."

File: nav2_dijkstra_planner/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+Dijkstra’s algorithm is guaranteed to find the shortest path under any condition.
+
+## Status
+Currently, DijkstraPlanner is a direct port from ROS1 MoveBase [Navfn](http://wiki.ros.org/navfn) planner.
+
+The Navfn planner assumes a circular robot and operates on a costmap to find a minimum cost plan from a start point to an end point in a grid. The navigation function is computed with Dijkstra's algorithm. The strategy is based on [High-Speed Navigation Using
```
Comment (mjeronimo) [inline nav2_dijkstra_planner/README.md]: Doesn't need a paragraph break here since the topic is still Navfn.

File: nav2_system_tests/README.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
+# Testing
```
Comment (mjeronimo) [inline nav2_system_tests/README.md]: Typically, these titles have the same name as the directory that they are READMEs for. In this case, it should be "System Tests" or "Nav2 System Tests" (we should be consistent). Perhaps we should rename this directory to be "nav2_tests". Should also mention more strongly that unit tests don't go here.

File: nav2_dijkstra_planner/README.md (hunk 4 above)
Comment (orduno) [reply nav2_dijkstra_planner/README.md]: Thanks, I'll make the change.

File: nav2_dijkstra_planner/README.md (hunk 5 above)
Comment (orduno) [reply nav2_dijkstra_planner/README.md]: Thanks, I'll change this too.

File: nav2_dijkstra_planner/README.md (hunk 6 above)
Comment (orduno) [reply nav2_dijkstra_planner/README.md]: Will change this.

File: nav2_system_tests/README.md (hunk 7 above)
Comment (orduno) [reply nav2_system_tests/README.md]: Good point, I'll make a few changes to make it more clear.

**Pull request merged**